homeassistant/components/huawei_lte/__init__.py

```diff
@@ -387,6 +391,40 @@ def _update_router(*_: Any) -> None:
     for router_config in config.get(DOMAIN, []):
         domain_config[url_normalize(router_config.pop(CONF_URL))] = router_config
 
+    def service_handler(service) -> None:
+        """Apply a service."""
+        url = service.data.get(CONF_URL)
+        routers = hass.data[DOMAIN].routers
+        if url:
+            router = routers.get(url)
+        else:
+            if len(routers) == 1:
+                router = next(iter(routers.values()))
+            else:
+                _LOGGER.error(
+                    "%s: more than one router configured, must specify one of URLs %s",
+                    service.service,
+                    sorted(routers.keys()),
```

Comment:
Defaulter iterator of a dict is the keys.
```suggestion
                    sorted(routers),
```